app/main.py

```diff
@@ -1,8 +1,22 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people_list: list) -> list:
+    person_instances = []
+    for person_dict in people_list:
+        person_instances.append(Person(name=person_dict["name"],
+                                       age=person_dict["age"]))
+
+    for person_dict in people_list:
+        current_person = Person.people[person_dict["name"]]
+        if "wife" in person_dict and person_dict["wife"]:
+            current_person.wife = Person.people[person_dict["wife"]]
+        if "husband" in person_dict and person_dict["husband"]:
+            current_person.husband = Person.people[person_dict["husband"]]
```

Comment:
The code assumes that the 'wife' and 'husband' keys in the `person_dict` will always have valid names that exist in `Person.people`. It would be safer to check if the name exists in `Person.people` before trying to access it to avoid potential `KeyError` exceptions.